reinforcement_learning/rlclientlib/ranking_event.h

```diff
@@ -33,38 +41,45 @@ namespace reinforcement_learning {
     ranking_event(ranking_event&& other);
     ranking_event& operator=(ranking_event&& other);
 
-    virtual void serialize(utility::data_buffer& buffer) override;
-  
+    flatbuffers::Offset<RankingEvent> serialize_eventhub_message(flatbuffers::FlatBufferBuilder& builder);
+    std::string str();
+
   public:
-    static ranking_event choose_rank(utility::data_buffer& oss, const char* event_id, const char* context,
+    static ranking_event choose_rank(const char* event_id, const char* context,
       unsigned int flags, const ranking_response& resp, float pass_prob = 1);
 
   private:
-    ranking_event(const char* event_id, float pass_prob, const std::string& body);
+    ranking_event(const char* event_id, bool deferred_action, float pass_prob, const char* context, const ranking_response& response);
 
-  private:
-    std::string _body;
+    std::string _context;
+    std::vector<uint64_t> _a_vector;
+    std::vector<float> _p_vector;
+    std::string _model_id;
+    bool _deferred_action;
   };
 
   //serializable outcome event
   class outcome_event : public event {
   public:
     outcome_event();
+
     outcome_event(outcome_event&& other);
     outcome_event& operator=(outcome_event&& other);
 
-    virtual void serialize(utility::data_buffer& buffer) override;
+    std::string str();
```

Comment:
a) to_json_string might be a better name
b) Please see comment on FB serialization knowledge living outside this class.  This method could benefit from a similar mechanism.